gprofiler/profilers/perf.py

```diff
@@ -202,11 +210,50 @@ def snapshot(self) -> ProcessToProfileData:
         for perf in self._perfs:
             perf.switch_output()
 
-        return {
-            # TODO generate appids for non runtime-profiler processes here
-            k: ProfileData(v, None, None)
-            for k, v in merge.merge_global_perfs(
-                self._perf_fp.wait_and_script() if self._perf_fp is not None else None,
-                self._perf_dwarf.wait_and_script() if self._perf_dwarf is not None else None,
-            ).items()
-        }
+        profile_data_dict = {}
+
+        for k, v in merge.merge_global_perfs(
+            self._perf_fp.wait_and_script() if self._perf_fp is not None else None,
+            self._perf_dwarf.wait_and_script() if self._perf_dwarf is not None else None,
+        ).items():
+            metadata = None
+            try:
+                if self._golang_metadata.is_golang_process(Process(k)):
+                    metadata = self._golang_metadata.make_application_metadata(Process(k))
+            except (NoSuchProcess, FileNotFoundError):
```

Comment:
`FileNotFoundError`?

Reply:
When trying to access `/proc/{pid}/exe`

Reply:
Although I now see `get_metadata` catches anything other than `NoSuchProcess`. Do you think I should catch `FileNotFoundError` and raise `NoSuchProcess` instead?